### PR TITLE
Support additional parquet compression codecs.  Fixes #714

### LINKED ISSUE #714: Support additional formats (eg, lz4) for parquet compression.
This will likely need build support as we need the C++ implementation of the compression codecs to avoid not-JITed-yet slowness in read/write operations in a cold JVM

### PR DESCRIPTION
**Pull request by jcferretti**

### CODE REVIEW

File: Integrations/build.gradle [hunk 1]
```diff
@@ -87,7 +87,9 @@ task deephavenPythonDockerfile(type: Dockerfile) {
     copyFile(new Dockerfile.CopyFile('/usr/src/app/dist', '.').withStage('deephaven-jpy-wheel'))
     copyFile(new Dockerfile.CopyFile('/usr/src/app/dist', '.').withStage('deephaven-wheel'))
 
-    runCommand('''set -eux;\\
```
Comment (jcferretti) [inline Integrations/build.gradle:90]: @devinrsmith There is probably a better place/way to do this, let me know what is best here.

File: build.gradle [hunk 2]
```diff
@@ -25,6 +25,8 @@ import java.util.concurrent.TimeUnit
 allprojects {
     repositories {
         mavenCentral()
+        maven { url 'https://jitpack.io' }
+        maven { url 'https://maven.twttr.com' }
```
Comment (devinrsmith) [inline build.gradle:29]: We should move these additional repos to ParquetHadoop/build.gradle if they are truly necessary.

We should add comments about which repositories we are expecting to get our non-standard dependencies from.

File: ParquetHadoop/build.gradle [hunk 3]
```diff
@@ -36,4 +35,8 @@ dependencies {
      *    [...]
      */
     runtimeOnly('com.fasterxml.woodstox:woodstox-core:5.2.1')
+    runtimeOnly('org.apache.hadoop.thirdparty:hadoop-shaded-guava:1.1.1')
+    runtime('com.hadoop.gplcompression:hadoop-lzo:0.4.20')
+    runtimeOnly('org.lz4:lz4-java:1.8.0')
+    runtimeOnly('com.github.rdblue:brotli-codec:0.1.1')
```
Comment (devinrsmith) [inline ParquetHadoop/build.gradle:41]: Are these dependencies that hadoop-common depends on? I'm worried that the transitive = false is biting us now.

Are these dependencies necessary; or optional, depending on how the data is encoded?

File: ParquetHadoop/build.gradle [hunk 4]
```diff
@@ -36,4 +35,8 @@ dependencies {
      *    [...]
      */
     runtimeOnly('com.fasterxml.woodstox:woodstox-core:5.2.1')
+    runtimeOnly('org.apache.hadoop.thirdparty:hadoop-shaded-guava:1.1.1')
+    runtime('com.hadoop.gplcompression:hadoop-lzo:0.4.20')
```
Comment (devinrsmith) [inline ParquetHadoop/build.gradle:39]: Why runtime instead of runtimeOnly?

File: Integrations/build.gradle (hunk 1 above)
Comment (devinrsmith) [reply Integrations/build.gradle:90]: This is an OK place to do it. I'm not a fan of adding universal dependencies to our base docker images; but liblzo2-2 is very small. The better place is *probably* py/jpy/Dockerfile under the runtime_reqs heading (deephaven/java-and-python). (It is a little weird that this file is currently under the jpy heading - we could refactor in the future, but I'm not too worried about that part.)

At the very least, we need to add a comment about why we are adding liblzo2-2.

**Review by devinrsmith — COMMENTED** [review]
Couple comments. In general, I'm concerned about the proper management and provenance of new dependencies; also, specifically for dependencies that may be optional.

File: ParquetHadoop/build.gradle (hunk 3 above)
Comment (jcferretti) [reply ParquetHadoop/build.gradle:41]: When we write a table in parquet format, they are pulled in dynamically (runtime dependency) depending on the compression codec specified for writing.  Traditionally we had supported only SNAPPY has a codec, but this diff is adding support to switch the default codec and LZ4, LZO, GZIP, BROTLI and ZSTD.
A later diff will make the compression codec part of an optional argument part of type ParquetWriteInstructions, that can give general instructions for writing (including other things like eg, remap column names).

File: ParquetHadoop/build.gradle (hunk 4 above)
Comment (jcferretti) [reply ParquetHadoop/build.gradle]: Testing leftover, not needed, fixed.

**Comment by jcferretti** [comment]
Added test cases.

File: Parquet/Parquet.gradle [hunk 5]
```diff
@@ -12,33 +16,19 @@ sourceSets {
...
-        // or replace transitive=false here w/ more exclusions;
-        // (we want to avoid pulling in netty, loggers, jetty-util, guice and asm).
-
-    }
+    implementation project(':CompilerTools')
 
-    runtime 'commons-collections:commons-collections:3.2.2'
-    runtime 'commons-logging:commons-logging:1.2'
+    runtimeOnly 'commons-collections:commons-collections:3.2.2'
+    //runtimeOnly 'commons-logging:commons-logging:1.2'
```
Comment (devinrsmith) [inline Parquet/Parquet.gradle:25]: We should move this next to the other runtimeOnly dependencies in ParquetHadoop/build.gradle.

```gradle
    runtimeOnly('com.fasterxml.woodstox:woodstox-core:5.2.1')
    runtimeOnly('org.apache.hadoop.thirdparty:hadoop-shaded-guava:1.1.1')
```

**Review by devinrsmith — APPROVED** [review]

**Pull request merged**